pulp_puppet_plugins/test/unit/test_install_distributor.py

```diff
@@ -622,12 +622,15 @@ def test_calls_rmtree(self, mock_rmtree):
 
         mock_rmtree.assert_called_once_with(self.path)
 
-    @mock.patch.object(installdistributor._LOGGER, 'error', spec_set=True)
+    @mock.patch.object(installdistributor._LOGGER, 'warn', spec_set=True)
     @mock.patch('shutil.rmtree', spec_set=True, side_effect=TestException)
-    def test_rmtree_exception(self, mock_rmtree, mock_error):
-        self.assertRaises(TestException, self.distributor.distributor_removed, self.repo, self.config)
-        # make sure the error was logged
-        self.assertEqual(mock_error.call_count, 1)
+    def test_rmtree_no_exception(self, mock_rmtree, mock_warn):
+        """Test that exception raised by shutil.rmtree won't be re-raised."""
+        try:
+            self.distributor.distributor_removed(self.repo, self.config)
+        except TestException:
+            self.fail('No exception should be raised.')
```

Comment:
I don't think the try/except is necessary. If it does raise, the test will Error.